### PR TITLE
Fix --enable-mca-dso with cuda

### PR DESCRIPTION
**Pull request by wckzhang**
Also fixed whitespace in opal/Makefile.am

### CODE REVIEW

**Comment by wckzhang** [comment]
Resolves: 9762

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/065d0a0bb7103355956e88c983bb45e8

**Comment by ibm-ompi** [comment]
The IBM CI (GNU/Scale) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/337c9fa96420f8496535941bbb6f600d

**Comment by ibm-ompi** [comment]
The IBM CI (PGI) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ca5a94dd2c4171987162eaf4dbb595aa

File: opal/Makefile.am [hunk 1]
```diff
@@ -91,6 +91,14 @@ lib@OPAL_LIB_NAME@_la_LDFLAGS = -version-info @libopen_pal_so_version@ \
 	$(opal_hwloc_LDFLAGS) \
 	$(opal_pmix_LDFLAGS)
 
+if OPAL_cuda_support
+if MCA_BUILD_opal_DSO
+lib@OPAL_LIB_NAME@_la_LIBADD += \
+	mca/common/cuda/lib@OPAL_LIB_NAME@mca_common_cuda.la
```
Comment (bwbarrett) [inline opal/Makefile.am:97]: This isn't right.  If the code is in mca/common, it's a common library and can't be special cased like this.  If it needs to be in libopen-pal.{a,so} all the time (as this patch suggests), then it doesn't belong in a common component and belongs in the base library.

**Review by bwbarrett — CHANGES_REQUESTED** [review]

File: opal/Makefile.am (hunk 1 above)
Comment (wckzhang) [reply opal/Makefile.am:97]: Hmm, I can revert the moving of code from datatype -> common then if that's the case

Comment (bwbarrett) [reply opal/Makefile.am:97]: William and I talked about this offline, but some notes for posterity.  There are some corner cases around common code in the core library that aren't worth fixing, so we talked about two possibilities:

1) If the problem is that it's not datatype code, then move it to another convenience library outside of common.  Like util or cuda or gpu or something at the top level of opal.

2) If the problem is you want to create an optional interface that is used by the core opal/ompi libraries, then you have to make a new framework, not use common.  The "must always be present" code then goes in the framework base, and the

**Pull request closed without merging**